tests/support/pytest/ldap.py

```diff
@@ -0,0 +1,392 @@
+
+    base: str = "dc=example,dc=org"
+    password: str = "adminpassword"
+    port: int = attr.ib(default=attr.Factory(ports.get_unused_localhost_port))
+    user: str = "admin"
+
+    @property
+    def userdn(self) -> str:
+        return f"cn={self.user},{self.base}"
+
+    @property
+    def uri(self) -> str:
+        """LDAP URI for use inside and outside the container.
+
+        This URI works both inside and outside because the network mode is set
+        to "host".
+        """
+        return f"ldap://localhost:{self.port}"
+
+    @property
+    def connect_spec(self):
+        return {
+            "url": self.uri,
+            "bind": {
+                "method": "simple",
+                "dn": self.userdn,
+                "password": self.password,
+            },
+        }
+
+    @classmethod
+    def default_config(cls, *args, **kwargs):
+        cfg = super().default_config(*args, **kwargs)
+
+        # Requirements on the minion daemon user:
+        #
+        #   * The `RUNTIME_VARS.CODE_DIR` directory must be readable by the
+        #     minion daemon user, otherwise Python will fail to import required
+        #     modules during minion start-up.
+        #
+        #   * As of 2022-10-05, the minion code requires the minion daemon user
+        #     to have an entry in the password database (`/etc/passwd`).
+        #
+        # The current user (`os.getuid()`) can access `RUNTIME_VARS.CODE_DIR`,
+        # but is unlikely to have an entry in the Docker container's password
+        # database.  We could pick a user that is known to exist in the image,
+        # but that user might not have access to `RUNTIME_VARS.CODE_DIR`.  The
+        # easiest way to ensure that both requirements are satisfied is to run
+        # the minion as root.  If the password database entry requirement is
+        # removed, the minion can be run as the current user by replacing
+        # `"root"` here with `os.getuid()` and changing `cmdline()` to pass
+        # `["--user", f"{os.getuid()}:{os.getgid()}"]`.
+        cfg.setdefault("user", "root")
+
+        return cfg
+
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
+        self.container_run_kwargs.setdefault("environment", {}).update(
+            {
+                "LDAP_ADMIN_PASSWORD": self.password,
+                "LDAP_ADMIN_USERNAME": self.user,
+                "LDAP_PORT_NUMBER": self.port,
+                "LDAP_ROOT": self.base,
+            },
+        )
+        # Note that the target directories have the same names as the source
+        # directories.  This avoids the need to translate path names.
+        self.container_run_kwargs.setdefault("volumes", {}).update(
+            {
+                # Bind mount the checked-out source code into the Docker
+                # container so that a minion daemon started inside the container
+                # will run the code to be tested.
+                RUNTIME_VARS.CODE_DIR: {
```

Comment:
Don't use `RUNTIME_VARS`, that's something we inherited from the old test suite.
```python
from tests.conftest import CODE_DIR
```